web/modules/custom/joinup_subscription/joinup_subscription.routing.yml

```diff
@@ -9,3 +9,31 @@ joinup_subscription.subscription_settings:
     parameters:
       user:
         type: entity:user
+
+joinup_subscription.subscription_settings_page:
+  path: '/user/subscription-settings'
+  defaults:
+    _controller: '\Drupal\joinup_subscription\Controller\SubscriptionSettings::subscriptionSettingsPage'
+    _title: 'Subscription settings'
+  requirements:
+    _user_is_logged_in: 'TRUE'
+
+joinup_subscription.subscriptions:
+  path: '/user/{user}/subscriptions'
+  defaults:
+    _form: '\Drupal\joinup_subscription\Form\SubscriptionDashboardForm'
+    _title: 'Collection subscriptions'
+  requirements:
+    _custom_access: '\Drupal\joinup_subscription\Controller\SubscriptionSettings::access'
+  options:
+    parameters:
+      user:
+        type: entity:user
+
+joinup_subscription.subscriptions_page:
+  path: '/user/subscriptions'
+  defaults:
+    _controller: '\Drupal\joinup_subscription\Controller\SubscriptionSettings::subscriptionDashboardPage'
```

Comment:
This is done to improve UX and follows the pattern used by core for pages related to the user account. See for example `UserController::userPage()`.

Reply:
I'm not questioning the reason why `/user/subscriptions` exists. I'm questioning the way is implemented. Just because it follows a core case doesn't mean it's the best implementation. There a useless redirect and a useless controller.